### PR TITLE
feat(comp:time-picker): add input functionality

### PR DESCRIPTION
**Pull request by sallerli1**
timepicker value can now be changed by input

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/IDuxFE/idux/blob/main/packages/site/src/docs/Contributing.zh.md#commit)
- [ ] Tests for the changes have been added/updated or not needed
- [ ] Docs and demo have been added/updated or not needed

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Component style update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
时间选择框不支持手动输入时间
时间选择面板交互为手动点击选择
image

## What is the new behavior?
增加手动输入时间的功能
时间选择面板交互修改为滚动选择+点击选择
image

## Other information
抽离了输入框与面板的输入控制模块、commonBinding以及输入框与面板的渲染方法封装

### CODE REVIEW

File: packages/components/time-picker/src/panel-column/PanelCell.tsx [hunk 1]
```diff
@@ -5,20 +5,22 @@
...
   setup(props) {
+    const { mergedPrefixCls } = inject(timePickerToken)!
     const classes = computed(() => {
       return {
-        'ix-time-picker-panel-cell': true,
-        'ix-time-picker-panel-cell-disabled': props.disabled,
-        'ix-time-picker-panel-cell-selected': props.selected,
+        [`${mergedPrefixCls.value}-panel-cell`]: true,
+        [`${mergedPrefixCls.value}-panel-cell-disabled`]: props.disabled,
+        [`${mergedPrefixCls.value}-panel-cell-selected`]: props.selected,
```
Comment (danranVm) [inline packages/components/time-picker/src/panel-column/PanelCell.tsx:23]:
```suggestion
        [prefixCls]: true,
        [`${prefixCls}-disabled`]: props.disabled,
        [`${prefixCls}-selected`]: props.selected,
```
const prefixCls = `${mergedPrefixCls.value}-panel-cell`

File: packages/components/time-picker/src/hooks.ts [hunk 2]
```diff
@@ -19,38 +20,46 @@ import { useValueAccessor } from '@idux/cdk/forms'
...
+  handleClose: () => void
+  handleInputConfirm: () => void
   focus: (options?: FocusOptions) => void
   blur: () => void
 }
 
-export function useCommonBindings<T extends TimePickerProps | TimeRangePickerProps>(
-  props: T,
-): CommonBindings<T extends TimeRangePickerProps ? [Date, Date] : Date> {
+export function useCommonBindings(
```
Comment (danranVm) [inline packages/components/time-picker/src/hooks.ts:43]: 这个函数的功能太重了，可以考虑拆分一下。

File: packages/components/time-picker/src/UseTimePickerRender.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  accessor: FormAccessor<Date>
+  isDisabled: ComputedRef<boolean>
+  renderInput: (attrs?: Record<string, unknown>, slots?: Slots) => VNodeTypes
+  renderPanel: (attrs?: Record<string, unknown>) => VNodeTypes
+  focus: (options?: FocusOptions | undefined) => void
+  blur: () => void
+  handleClose: () => void
+}
+
+export function useTimePickerRender(
```
Comment (danranVm) [inline packages/components/time-picker/src/UseTimePickerRender.tsx:33]: 这里其实就是两个组件的 render 函数，抽成 composable 感觉挺奇怪的。

File: packages/components/time-picker/demo/Basic.vue [hunk 4]
```diff
@@ -9,7 +9,7 @@
 import { ref } from 'vue'
 
 function changeToNow() {
-  time.value = Date.now()
+  time.value = new Date()
```
[the remark this answers is not in the document]
Comment (sallerli1) [reply packages/components/time-picker/demo/Basic.vue:12]: 这个是因为value的类型定义的是Date，Date.now()能正常运行但是会有warning

File: packages/components/time-picker/src/hooks.ts (hunk 2 above)
Comment (sallerli1) [reply packages/components/time-picker/src/hooks.ts:43]: 这个结合日期选择整体再设计之后可能就不需要了，现在还不知道怎么改

File: packages/components/time-picker/src/UseTimePickerRender.tsx (hunk 3 above)
Comment (sallerli1) [reply packages/components/time-picker/src/UseTimePickerRender.tsx:33]: 这个API的意义在于把比较繁琐的控制联动逻辑还有属性时间的绑定封装起来，结合两个组件但同时又不限制他们实际的组件结构。（实际上是为了好复用代码后面实现输入框在面板里面的交互方式）

**Pull request closed without merging**